mne/stats/regression.py

```diff
@@ -392,3 +392,254 @@ def _make_evokeds(coefs, conds, cond_length, tmin_s, tmax_s, info):
+    """Return a time-lagged input timeseries.
+
+    Parameters
+    ----------
+    ts: array, shape (n_feats, n_times)
+        The timeseries to delay
+    sfreq: int
+        The sampling frequency of the series
+    delays: list of floats
+        The time (in seconds) of each delay. Negative means
+        timepoints in the past, positive means timepoints in
+        the future.
+
+    Returns
+    -------
+    delayed: array, shape(n_feats * n_delays, n_times)
+        The delayed matrix
+    """
+    delayed = np.zeros([len(delays), ts.shape[-1]])
+    for ii, delay in enumerate(delays):
+        roll_amount = -1 * int(delay * sfreq)
+        rolled = np.roll(ts, roll_amount, axis=-1)
+        if roll_amount < 0:
+            rolled[:, roll_amount:0] = 0
+        elif roll_amount > 0:
+            rolled[:, 0:roll_amount] = 0
+        delayed[ii] = rolled
+    delayed = np.vstack(delayed)
+    return delayed
+
+
+def _events_to_continuous(raw, events, event_id):
+    """Turn an MNE events array into a binary matrix of (n_times, n_events)."""
+    # Prep events for this raw instance and pull unique event types
+    if events.ndim != 2:
+        raise ValueError("events must be shape (n_trials, 3),"
+                         " found shape %s" % events.shape)
+    events = events.copy()
+    events[:, 0] -= raw.first_samp
+    unique_event_types = np.unique(events[:, 2])
+
+    # Iterate through event types and create columns of event onsets for each.
+    events_continuous = np.zeros([len(unique_event_types), raw.n_times])
+    for ii, ev_type in enumerate(unique_event_types):
+        msk_events = events[:, 2] == ev_type
+        i_ev_rows = events[msk_events, 0]
+        events_continuous[ii, i_ev_rows] = 1
+
+    # Handle event names
+    if event_id is None:
+        event_names = ['event_%s' % ii
+                       for ii in range(len(unique_event_types))]
+    else:
+        event_names = [event_id[ii] for ii in unique_event_types]
+    return events_continuous, event_names
+
+
+def _check_estimator(est):
+    from sklearn.linear_model import Ridge
+    from sklearn.pipeline import Pipeline
+    if est is None:
+        est = Ridge()
+    elif isinstance(est, string_types):
+        if est not in estimator_dict.keys():
+            raise ValueError("No such solver: {0}".format(est))
+        est = estimator_dict[est]
+
+    # Make sure we have a pipeline
+    if not isinstance(est, Pipeline):
+        est = Pipeline([('est', est)])
+    for imethod in ['fit', 'predict']:
+        if not hasattr(est.steps[-1][-1], imethod):
+            raise ValueError('estimator must have a %s method' % imethod)
+    return est
+
+
+# Custom string-supported estimators
+class Cholesky(object):
+    def fit(self, X, y):
+        a = (X.T * X).toarray()  # dot product of sparse matrices
+        self.coef_ = linalg.solve(a, X.T * y.T, sym_pos=True,
+                                  overwrite_a=True, overwrite_b=True).T
```

Comment:
Factor out this https://github.com/mne-tools/mne-python/blob/master/mne/stats/regression.py#L248
and use it instead